### PR TITLE
chore(nsconfig): export constants and getDefautNConfig to public API

### PR DESCRIPTION
**Pull request by KristianDD**

### CODE REVIEW

**Comment by KristianDD** [comment]
Ping @rosen-vladimirov @Mitko-Kerezov

File: lib/services/project-data-service.ts [hunk 1]
```diff
@@ -124,5 +124,13 @@ export class ProjectDataService implements IProjectDataService {
 			projectFilePath
 		};
 	}
+
+	@exported("projectDataService")
+	public getNsConfigDefaultContent(data: Object): string {
```
Comment (rosen-vladimirov) [inline lib/services/project-data-service.ts]: can you please add the method to the lib tests: https://github.com/NativeScript/nativescript-cli/blob/master/test/nativescript-cli-lib.ts#L19

File: lib/project-data.ts [hunk 2]
```diff
@@ -169,6 +169,10 @@ export class ProjectData implements IProjectData {
 		return this.$fs.readText(configNSFilePath);
 	}
 
+	public getNsConfigRelativePath() {
```
Comment (Fatme) [inline lib/project-data.ts]: Missing return type
```
public getNsConfigRelativePath(): string { ... }
```

File: lib/services/project-data-service.ts [hunk 3]
```diff
@@ -124,5 +124,13 @@ export class ProjectDataService implements IProjectDataService {
...
+		const config: INsConfig = {};
+		Object.assign(config, data);
+
+		return JSON.stringify(config);
```
Comment (Fatme) [inline lib/services/project-data-service.ts]: Can we use `return JSON.stringify(data);` If it is possible `config` will not be needed.

File: PublicAPI.md [hunk 4]
```diff
@@ -187,6 +188,17 @@ Returns an IProjectData object that is initialized with the provided package.jso
 getProjectDataFromContent(packageJsonContent: string, nsconfigContent: string, projectDir?: string): IProjectData
 ```
 
+### getNsConfigDefaultContent
+Returns the default content of "nsconfig.json" merged with the properties provided by the the `data` argument.
+* Definition:
+```TypeScript
+/**
+ * Returns the default content of "nsconfig.json" merged with the properties provided by the the "data" argument.
```
Comment (sis0k0) [inline PublicAPI.md]: typo: double `the`

File: PublicAPI.md [hunk 5]
```diff
@@ -187,6 +188,17 @@ Returns an IProjectData object that is initialized with the provided package.jso
...
+ * @param {string} data Properties that should not be defaulted.
```
Comment (sis0k0) [inline PublicAPI.md]: ` * @param {string} `, isn't the type `Object`?

File: PublicAPI.md (hunk 4 above)
Comment (sis0k0) [inline PublicAPI.md]: double `the`
`provided by the the "data"` -> `provided by the "data"`

File: lib/project-data.ts [hunk 6]
```diff
@@ -169,6 +169,10 @@ export class ProjectData implements IProjectData {
...
+		return  constants.CONFIG_NS_FILE_NAME;
```
Comment (sis0k0) [inline lib/project-data.ts]: There are two white spaces between `return` and `constants.`

File: lib/services/project-data-service.ts (hunk 3 above)
Comment (KristianDD) [reply lib/services/project-data-service.ts]: The main idea of the method is to merge default values with some overrides. For now nsconfig's default state is an empty object but writing the method this way makes it a bit more readable.

**Comment by KristianDD** [comment]
run ci

**Comment by KristianDD** [comment]
run ci

**Pull request merged**